pkg/operator/upgradebackupcontroller/upgradebackupcontroller.go

```diff
@@ -0,0 +1,258 @@
+
+	// Backup has completed or failed, update status accordingly
+	_, _, err = v1helpers.UpdateStatus(c.operatorClient, v1helpers.UpdateConditionFn(operatorv1.OperatorCondition{
+		Type:   "RecentBackup",
+		Status: backupStatus,
+		Reason: backupVersion,
+	}))
+	return err
+}
+
+func ensureRecentBackup(ctx context.Context, kubeClient kubernetes.Interface, podLister corev1listers.PodLister, nodeLister corev1listers.NodeLister, clusterBackupVersion, targetImagePullSpec string, recorder events.Recorder) (operatorv1.ConditionStatus, error) {
+	delete := func(pod *corev1.Pod) error {
+		err := kubeClient.CoreV1().Pods(operatorclient.TargetNamespace).Delete(ctx, pod.Name, metav1.DeleteOptions{Preconditions: metav1.NewUIDPreconditions(string(pod.UID))})
+		if err != nil {
+			return err
+		}
+		return nil
+	}
+
+	backupPods, err := podLister.List(labels.Set{"app": "cluster-backup"}.AsSelector())
+	if err != nil {
+		return "", err
+	}
+
+	var errs []error
+	for _, pod := range backupPods {
+		if pod.Labels[clusterBackupVersionLabel] != clusterBackupVersion {
+			// Delete pods with invalid version
+			if err := delete(pod); err != nil {
+				errs = append(errs, err)
+			}
+			continue
+		}
+
+		switch pod.Status.Phase {
+		case corev1.PodSucceeded:
+			return operatorv1.ConditionTrue, nil
+		case corev1.PodFailed:
+			if err := delete(pod); err != nil {
+				klog.Errorf("failed to delete pod %s/%s: %v", pod.Namespace, pod.Name, err)
+			}
+			return operatorv1.ConditionFalse, nil
+		default:
+			return operatorv1.ConditionUnknown, nil
+		}
+	}
+
+	node, err := getNodeWithAdequateAllocatableStorage(ctx, podLister, nodeLister)
+	if err != nil {
+		return "", err
+	}
+
+	_, _, err = manageRecoveryPod(ctx, node.Name, targetImagePullSpec, clusterBackupVersion, kubeClient.CoreV1(), recorder)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("%q: %v", "pod/cluster-backup", err))
+	}
+
+	if len(errs) > 0 {
+		return "", utilerrors.NewAggregate(errs)
+	}
+
+	return operatorv1.ConditionUnknown, nil
+}
+
+func manageRecoveryPod(ctx context.Context, nodeName, targetImagePullSpec, clusterBackupVersion string, client coreclientv1.PodsGetter, recorder events.Recorder) (*corev1.Pod, bool, error) {
+	pod := resourceread.ReadPodV1OrDie(etcd_assets.MustAsset("etcd/cluster-backup-pod.yaml"))
+	pod.Name = "cluster-backup-" + nodeName
+	pod.Namespace = operatorclient.TargetNamespace
+	pod.Spec.NodeName = nodeName
+	pod.Labels = map[string]string{
+		"app":                     "cluster-backup",
+		clusterBackupVersionLabel: clusterBackupVersion,
+	}
+	pod.Spec.Containers[0].Image = targetImagePullSpec
+
+	return resourceapply.ApplyPod(ctx, client, recorder, pod)
+}
+
+// isRequireRecentBackup checks the conditions of ClusterVersion to verify if a backup is required.
+func isRequireRecentBackup(config *configv1.ClusterVersion) bool {
+	for _, condition := range config.Status.Conditions {
+		// Check if Failing is true and Message field containers the string RecentBackup.
```

Comment:
Too much time in containers ;) 

Unless it's a joke, in which case I am all for leaving it! :D 

```suggestion
		// Check if Failing is true and Message field contains the string RecentBackup.
```